lib/internal/modules/cjs/loader.js

```diff
@@ -573,28 +574,33 @@ Module._resolveFilename = function(request, parent, isMain, options) {
 
   var paths;
 
-  if (typeof options === 'object' && options !== null &&
-      Array.isArray(options.paths)) {
-    const isRelative = request.startsWith('./') || request.startsWith('../') ||
-        (isWindows && request.startsWith('.\\') || request.startsWith('..\\'));
+  if (typeof options === 'object' && options !== null) {
+    if (Array.isArray(options.paths)) {
+      const isRelative = request.startsWith('./') ||
+          request.startsWith('../') ||
+          (isWindows && request.startsWith('.\\') ||
+          request.startsWith('..\\'));
 
-    if (isRelative) {
-      paths = options.paths;
-    } else {
-      const fakeParent = new Module('', null);
+      if (isRelative) {
+        paths = options.paths;
+      } else {
+        const fakeParent = new Module('', null);
 
-      paths = [];
+        paths = [];
 
-      for (var i = 0; i < options.paths.length; i++) {
-        const path = options.paths[i];
-        fakeParent.paths = Module._nodeModulePaths(path);
-        const lookupPaths = Module._resolveLookupPaths(request, fakeParent);
+        for (var i = 0; i < options.paths.length; i++) {
+          const path = options.paths[i];
+          fakeParent.paths = Module._nodeModulePaths(path);
+          const lookupPaths = Module._resolveLookupPaths(request, fakeParent);
 
-        for (var j = 0; j < lookupPaths.length; j++) {
-          if (!paths.includes(lookupPaths[j]))
-            paths.push(lookupPaths[j]);
+          for (var j = 0; j < lookupPaths.length; j++) {
+            if (!paths.includes(lookupPaths[j]))
+              paths.push(lookupPaths[j]);
+          }
         }
       }
+    } else if (options.paths !== undefined) {
+      throw new ERR_INVALID_OPT_VALUE('paths', options.paths);
```

Comment:
```suggestion
      throw new ERR_INVALID_ARG_TYPE('options.paths', 'Array', options.paths);
```

This way it's also detected that it's a property.